guides/common/modules/proc_registering-project-as-a-keycloak-client.adoc

```diff
@@ -0,0 +1,74 @@
+[id="registering-project-as-a-keycloak-client_{context}"]
+= Registering {Project} as a {Keycloak} Client
+Use this procedure to register {Project} to {Keycloak} as a client and configure {Project} to use {Keycloak} as an authentication source.
+
+You can configure {Project} and {Keycloak} with two different authentication methods:
+
+. Users authenticate to {Project} using the {Project} web UI.
+. Users authenticate to {Project} using the {Project} CLI.
+
+You must decide on how you want your users to authenticate in advance because both methods require different {Project} clients to be registered to {Keycloak} and configured.
+The steps to register and configure {Project} client in {Keycloak} are distinguished within the procedure.
+
+You can also register two different {Project} clients to {Keycloak} if you want to use both authentication methods and configure both client accordingly.
+
+.Procedure
+
+. On the {Project} server, install the following packages:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {package-install-project} mod_auth_openidc keycloak-httpd-client-install
+----
+
+. Register {Project} to {Keycloak} as a client.
+Note that you the registration process for logging in using the web UI and the CLI are different.
+You can register two clients {Project} clients to {Keycloak} to be able to log in to {Project} from the web UI and the CLI.
++
+* If you want you users to authenticate to {Project} using the web UI, create a client as follows:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# keycloak-httpd-client-install --app-name foreman-openidc \
+--keycloak-server-url "_https://{Keycloak-short}.com_" \
+--keycloak-admin-username "_admin_" \
+--keycloak-realm "_{Project}_Realm_" \
+--keycloak-admin-realm master \
+--keycloak-auth-role root-admin \
+-t openidc -l /users/extlogin --force
+----
++
+Enter the password for the administer account when prompted.
+This command creates a client for {Project} in {Keycloak}.
++
+Then, configure {Project} to use {Keycloak} as an authentication source:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# satellite-installer --foreman-keycloak true \
+--foreman-keycloak-app-name "foreman-openidc" \
+--foreman-keycloak-realm "_{Project}_Realm_"
+----
++
+* If you want your users to authenticate to {Project} using the CLI, create a client as follows:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# keycloak-httpd-client-install --app-name hammer-openidc \
+--keycloak-server-url "_https://{Keycloak-short}.com:8443_" \
+--keycloak-admin-username "_admin_" \
+--keycloak-realm "_{Project}_Realm_" \
+--keycloak-admin-realm master \
+--keycloak-auth-role root-admin \
+-t openidc --force
```

Comment:
why we are adding this in here ? why no in CLI block. Anyways it is missing the `-l /users/extlogin` option

Reply:
We dont need the /users/extlogin for the cli users, I think, can you test?

This command creates a apache file in the system with the endpoint but since we are using the CLI, we dont require a endpoint. Most important part in a CLI is the redirect URL.

Reply:
Is this something that you will be able to test @omkarkhatavkar?

Reply:
Okay, I mainly use the UI steps then override while testing with CLI. Let me try those steps.

Reply:
@spetrosi @rabajaj0509 It is failing with below error in case if we does not include the `-l /users/extlogin`

Error Received : 

`keycloak-httpd-client-install: error: You must specify -l/--protected-locations`

Reply:
Thanks for checking, @omkarkhatavkar, I have added `-l /users/extlogi` to the command.